File: venues/swsa.semanticweb.org/ISWC/2018/DeSemWeb/process/commentProcess.js
function(){
    var or3client = lib.or3client;

    var SHORT_PHRASE = 'DeSemWeb 2018';

    var origNoteP = or3client.or3request(or3client.notesUrl + '?id=' + note.forum, {}, 'GET', token);
    var replytoNoteP = note.replyto ? or3client.or3request(or3client.notesUrl + '?id=' + note.replyto, {}, 'GET', token) : null;

    Promise.all([
      origNoteP,
      replytoNoteP
    ]).then(function(result) {

      var origNote = result[0].notes[0];
      var replytoNote = note.replyto ? result[1].notes[0] : null;
      var replytoNoteSignatures = replytoNote ? replytoNote.signatures : [];
      var author_mail;

      var selfComment = replytoNoteSignatures.indexOf(note.signatures[0]) > -1 ? true : false;
      console.log('self comment detected');
      if(!selfComment){
        if(replytoNote.id == origNote.id){
          author_mail = {
            groups: origNote.content.authorids,
            subject: 'Your submission to ' + SHORT_PHRASE +  ' has received a comment',
            message: 'Your submission to ' + SHORT_PHRASE + ' has received a comment.\n\nComment title: ' + note.content.title + '\n\nComment: ' + note.content.comment + '\n\nTo view the comment, click here: ' + baseUrl+'/forum?id=' + note.forum +'&noteId='+note.id
          };
        } else {
          author_mail = {
            groups: replytoNote.signatures == '(anonymous)' ? [] : replytoNote.signatures,
            subject: 'Your comment has received a response',
            message: 'Your comment titled "'+replytoNote.content.title+'" has received a response.\n\nComment title: ' + note.content.title + '\n\nComment: ' + note.content.comment + '\n\nTo view the comment, click here: ' + baseUrl+'/forum?id=' + note.forum +'&noteId='+note.id
          };
        }
        promises = [or3client.or3request(or3client.mailUrl, author_mail, 'POST', token)];
      } else {
        promises = []
      }

      return Promise.all(promises);
    })
    .then(result => done())
    .catch(error => done(error));

    return true;
  };

File: venues/reproducibility-challenge.github.io/Reproducibility_Challenge/NeurIPS/2019/process/commentProcess.py
def process(client, note, invitation):
    SHORT_PHRASE = "NeurIPS Reproducibility Challenge"
    # get all notification tags for this paper
    notify_inv = invitation.id.replace('Comment', 'Notification_Subscription')
    # TODO change to iterget
    notifications = client.get_tags(invitation=notify_inv)

    email_list = [tag.signatures[0] for tag in notifications if tag.tag == 'Immediate']
    all_notifiers = [tag.signatures[0] for tag in notifications]

    # get submission author and add to email list if author hasn't set a notification tag
    forumNote = client.get_note(id=note.forum)

    # add comment author to email list if does not have a tag
    if note.signatures[0] and note.signatures[0] not in all_notifiers:
        email_list.append[note.signatures[0]]

    if email_list:
        # send email to those in the immediate notification group
        subject = '[' + SHORT_PHRASE + '] Paper Title: "' + forumNote.content['title'] + '" received a comment'
        formatted_msg = 'The NeurIPS submission titled "' + forumNote.content['title'] + '" has received a comment.\n\nComment title: ' \
                        + note.content['title'] + '\n\nComment: ' + note.content['comment'] + '\n\nTo view the comment, click here: ' + \
                        client.baseurl + '/forum?id=' + note.forum + '&noteId=' + note.id + '\n\nIf you wish to change your email notification preferences for comments on this paper, log into OpenReview.net, visit the link above and change the Notification Subscription frequency.'

        response = client.send_mail(subject, email_list, formatted_msg)
